microprofile/testing/testng/src/main/java/io/helidon/microprofile/testing/testng/ConfigBlob.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.testing.testng;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UncheckedIOException;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Objects;
+import java.util.Properties;
+
+import io.helidon.config.mp.MpConfigSources;
+import io.helidon.config.yaml.mp.YamlMpConfigSource;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * Defines the configuration as a String in {@link #content()} for the
+ * given type.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Inherited
+public @interface ConfigBlob {
+
+    /**
+     * Specifies the format type of the {@link #content()}.
+     *
+     * It defaults to {@link Type#PROPERTIES}.
+     *
+     * @return the supported type
+     */
+    Type type() default Type.PROPERTIES;
+
+    /**
+     * Configuration content.
+     *
+     * @return String with content.
+     */
+    String content();
+
+    /**
+     * Different formats of the configuration content.
+     */
+    enum Type {
```

Comment:
This effectively hard-codes the config sources that are usable.
Shouldn't we instead support a model that can lead to any config source implementation on the classpath ?

E.g. `io.helidon.config.mp.spi.MpMetaConfigProvider` has a method `supportedTypes` that matches providers to "types".

Reply:
I have taken a look into this. The problem is that what you have in classpath/modulepath will not necessarily match with the inputs of the user in the test.

You could have loaded YamlMetaConfigProvider and still the user sets in the test the next (in properties format):
```java
@AddConfigBlock( """
    foo=bar
    bar=foo
""")
```

I agree that we should have a common place for the `type`. For example in case in the future we add a new config support, like JSON or XML, we would need to update also the enum.

I can make the type as string. In this way, we can manage the string and hope that at some point we will have a common point to do this conversion.

Reply:
The snippet you've shown above does not specify the type. It should specify the type with something that we can match against loaded providers. I mentioned `MpMetaConfigProvider` as an example of a similar mechanism.

We need a way to create a config source based on a type, `MpMetaConfigProvider` demonstrates something similar to what we need but does not seem directly usable.

Either we need a new SPI, or update the MpConfig API to add the capability to specify config sources as a pairs of type and reader.

JSON is already supported, see `hocon-mp`.

Reply:
OK, so the task is going to be to create one SPI that will contain:
```
supportedTypes(): Collection<String>
create(Reader): ConfigSource

```
Then configure it in every module under /META-INF/services, iterate the ServiceLoaders to find which type is supported and invoke the create(Reader).

This can be done in a separate issue. In this task I can just set the `type` as string. Later we can use the type to do the process above.